crates/oci-distribution/src/errors.rs

```diff
@@ -8,9 +8,11 @@
 pub struct OciError {
     /// The error code
     pub code: OciErrorCode,
-    /// A message associated with the error
+    /// An optional message associated with the error
+    #[serde(default)]
     pub message: String,
-    /// Unstructured data associated with the error
+    /// Unstructured optional data associated with the error
+    #[serde(default)]
     pub detail: serde_json::Value,
 }
```

Comment:
The strategy of using serde defaulting to `Default::default()` is such that there's no cascading refactor on `krustlet`. However, if you are good with it, I can make `message` and `detail` be an `Option` that I believe is closer to the spec wording.